R/az_hourly.R

```diff
@@ -9,10 +9,13 @@
 #'   stations will be returned
 #' @param start_date_time A length 1 vector of class POSIXct or character in
 #'   YYYY-MM-DD HH format.  Will be rounded **down** to the nearest hour if more
-#'   precision is supplied.
+#'   precision is supplied. If only a date (YYYY-MM-DD) is supplied, data will
+#'   be requested starting at 01:00:00 of that day
 #' @param end_date_time A length 1 vector of class POSIXct or character in
 #'   YYYY-MM-DD HH format.  Will be rounded **down** to the nearest hour if more
-#'   precision is supplied.  Defaults to the current time if left blank.
+#'   precision is supplied.  If only a date (YYYY-MM-DD) is supplied, data will
+#'   be requested through the *end* of that day (23:59:59). Defaults to the
+#'   current time if left blank.
```

Comment:
`#' Defaults to the current date and time if left blank.`

Running this on 2023-10-23 just before noon:
`az_hourly(station_id = "az01", start_date_time = "2023-10-21")`

. . . returns all expected data through 10/23 1100, but gives:

`Warning message:
In az_hourly(station_id = "az01", start_date_time = "2023-10-21") :
  Some requested data were unavailable`

I suspect the message is due to the fact that the warning check is expecting a full day of hourly data for 10/23, which isn't the case here.

Although not critical, perhaps a new issue to fine tune the hourly warning message to check by date and hour instead of just date? It's a bit like the QA Dashboard check for stations not reporting hourly data, right?

Reply:
I'll investigate this.  This also makes me think we should print a message when only a date is supplied to `az_hourly()` so the user knows how it's being interpreted.

Reply:
Ah ha! It was a timezone issue.  I needed to set the timzone as MST when being parsed, otherwise at some point the start date was ending up in UTC and the end date was ending up in the local timezone.

Reply:
Timezone trickery returns! Something similar was tripping me up when working on the `wind_2min_timestamp` variable issue.